front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -113,4 +114,59 @@ const handleOperation = async ({
   }
 };
 
+const handleNodeOperation = ({
+  type,
+  node,
+  timeTableId,
+}: {
+  type: NGEEvent['type'];
+  node: Node;
+  timeTableId: number;
+}) => {
+  const { betriebspunktName: trigram, positionX, positionY } = node;
+  switch (type) {
+    case 'create':
+    case 'update': {
+      nodeStore.set(timeTableId, { trigram, positionX, positionY });
+      break;
+    }
+    case 'delete': {
+      nodeStore.delete(timeTableId, trigram);
+      break;
+    }
+    default:
+      break;
+  }
+};
+
+const handleOperation = async ({
+  event,
+  dispatch,
+  timeTableId,
+  netzgrafikDto,
+}: {
+  event: NGEEvent;
+  dispatch: AppDispatch;
+  timeTableId: number;
+  netzgrafikDto: NetzgrafikDto;
+}) => {
+  const { type } = event;
+  switch (event.objectType) {
+    case 'node':
+      handleNodeOperation({ type, node: event.node!, timeTableId });
```

Comment:
Once you have modified the NGEEvent, you can remove the `!` after `node` l156 and `trainrun` line161

Reply:
true !